src/writer.rs

```diff
@@ -794,18 +795,33 @@ impl<D: Distance> Writer<D> {
                 let error_snd = error_snd.clone();
                 let nb_items_progress = nb_items_progress.clone();
                 scope.spawn(move |s| {
-                    let ret = self.incremental_index_large_descendant(
-                        rng,
-                        options,
-                        &error_snd,
-                        nb_items_progress,
-                        s,
-                        (item_id, item_indices),
-                        frozen_reader,
-                        tmp_nodes,
-                    );
-                    if let Err(e) = ret {
-                        let _ = error_snd.try_send(e);
+                    let ret = std::panic::catch_unwind(AssertUnwindSafe(|| {
+                        self.incremental_index_large_descendant(
+                            rng,
+                            options,
+                            &error_snd,
+                            nb_items_progress,
+                            s,
+                            (item_id, item_indices),
+                            frozen_reader,
+                            tmp_nodes,
+                        )
+                    }));
+                    match ret {
+                        Ok(Ok(())) => (),
+                        Ok(Err(e)) => {
+                            let _ = error_snd.try_send(e);
+                        }
+                        Err(panic) => {
+                            let msg = match panic.downcast_ref::<&'static str>() {
```

Comment:
```suggestion
                            // In case of a panic, the error message is either a static `str` or a `String`. If it's something else, we don't know how to display it.
                            let msg = match panic.downcast_ref::<&'static str>() {
```